rnote-ui/src/canvas/mod.rs

```diff
@@ -554,6 +582,75 @@ impl RnoteCanvas {
         }
     }
 
+    pub(crate) fn create_output_file_monitor(&self, file: &gio::File, appwindow: &RnoteAppWindow) {
+        match file.monitor_file(gio::FileMonitorFlags::WATCH_MOVES, gio::Cancellable::NONE) {
+            Ok(output_file_monitor) => {
+                output_file_monitor.connect_changed(
+                    glib::clone!(@weak self as canvas, @weak appwindow => move |_monitor, file, other_file, event| {
+                        let dispatch_toast_reload_modified_file = || {
+                            appwindow.canvas().set_unsaved_changes(true);
+
+                            appwindow.dispatch_toast_w_button_singleton(&gettext("Opened file was modified on disk."), &gettext("Reload"), clone!(@weak appwindow => move |_reload_toast| {
+                                if let Some(output_file) = appwindow.canvas().output_file() {
+                                    if let Err(e) = appwindow.load_in_file(&output_file, None) {
+                                        log::error!("failed to reload current output file, {}", e);
+                                    }
+                                }
+                            }), 0, &mut canvas.imp().output_file_modified_toast_singleton.borrow_mut());
+                        };
+
+                        match event {
+                            gio::FileMonitorEvent::Changed => {
+                                dispatch_toast_reload_modified_file();
+                            },
+                            gio::FileMonitorEvent::Renamed => {
+                                // if previous file name was .goutputstream-<hash>, then the file has been replaced using gio.
+                                if FileType::is_goutputstream_file(&file) {
+                                    if canvas.output_file_expect_write() {
+                                        // => file has been modified due to own save, don't do anything.
+                                        canvas.set_output_file_expect_write(false);
```

Comment:
maybe it should also be cleared on the changed event or even always at the end of the changed callback? I am thinking about other platforms (maybe a future port to windows ) where the behaviour might be different. It makes me a bit anxious that clearing the flag is behind so many conditions.

Reply:
Right. I'd personally clear it after every event then, i.e. move it below line 636.

Reply:
yeah, I'm not confident enough that this always will be cleared. The worst thing that could happen is an invalid warning when a user changes the file externally while it is being written, but I don't think thats really problematic

Reply:
Yeah. On another note, I think the part after the OR should be removed in the snippet I sent above, right? I don't think that the output file is ever set to a `.goutputstream-` file, but I might be missing the thought that I had while I writing this.

Reply:
> We could even remove the condition here:

I think so too actually, recreating the monitor wouldn't be an issue, would it?

and also agree with always clearing it at the end of the callback

Reply:
I would remove this `if !canvas.output_file_expect_write() || !FileType::is_goutputstream_file(output_file)` entirely, because it could be an issue when saving the file for the first time. when only `!is_outputstream_file()` is removed the output monitor will never be created I thikn